services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_utils.py

```diff
@@ -173,9 +173,10 @@ async def service_save_state(
         size = await sidecars_client.save_service_state(
             scheduler_data.endpoint, progress_callback=progress_callback
         )
-    get_instrumentation(app).dynamic_sidecar_metrics.push_service_state_rate.labels(
-        **get_metrics_labels(scheduler_data)
-    ).observe(get_rate(size, duration.to_flaot()))
+    if rate := get_rate(size, duration.to_float()) >= 1:
```

Comment:
Q: you return `size` argument. I guess your `save_service_state` "knows" how many files it transfered. so why not get that information back? if it transferred 0 files, there is not rate. Because with this "hack" here it can very well backfire.

Reply:
I have a few questions here, since I'm a bit puzzled by this comment.
1. what is the difference between transferring 0 files and 1 file of size 0 ?
2. and what is the difference between transferring 0 files and 100 files of size 0 ?
3. What metric would you intend using here? Why would I return the number of transferred files? (I do not see how the number of transferred files is helpful)

--- 

And the hard part: to extract the number of files it becomes a bit of a nightmare.
- dy-sidecar has to return the number (not the hardest thing in the world)
- the data manager layer which is ~ 6 calls deep has to return this number for (states, outputs inputs)
  - states: returns nothing
  - outputs & inputs: return the file size (If I remember correctly)
- need to extract the total number of trasnferred files from the rclone command and send it back.

--- 

I think this brigs quite a lot of complication to the backend just to support this feature. Do you it's necessary to go through all of these changes?

What I am really asking is: are the benefits you state really that big that they are with this time investment?

Reply:
As I noted in my overall comment.
In case the transfer rate is between 0 and 1 for whatever reason, then it will not be reported.

I would at least prefer that you check whether the transferred size is equal to 0 or not instead. This is already wrong, but at least less than using the rate which is yet another layer of by-product.